File: Review-04/Example-5/src/test/java/edu/odu/cs/cs330/examples/TestBoard.java
package edu.odu.cs.cs330.examples;

import org.junit.FixMethodOrder;
import org.junit.runners.MethodSorters;
import org.junit.Test;
import org.junit.Before;

import static org.junit.Assert.*;

import static org.hamcrest.CoreMatchers.*;
import org.hamcrest.core.IsNull;

/**
 * 1 - Does this piece of code perform the operations
 *     it was designed to perform?
 *
 * 2 - Does this piece of code do something it was not
 *     designed to perform?
 *
 * 1 Test per mutator
 */
@FixMethodOrder(MethodSorters.NAME_ASCENDING)
public class TestBoard
{
    final String expectedEmptyStr = "1|2|3\n4|5|6\n7|8|9\n";

    Board aBoard;

    @Before
    public void setUp()
    {
        aBoard = new Board();
    }

    @Test
    public void testDefaultConstructor()
    {
        for (int i = 1; i <= 9; i++) {
            char expectedChar = Character.forDigit(i, 10);
            assertThat(aBoard.getCell(i), is(expectedChar));
        }

        assertThat(aBoard.toString(), equalTo(expectedEmptyStr));
        assertFalse(aBoard.isFull());

        Board.Pair[] retrieved = aBoard.get3Cells(1, 2, 3);
        Board.Pair[] expected = {new Board.Pair(0, '1'),
                                 new Board.Pair(1, '2'),
                                 new Board.Pair(2, '3')};

        assertThat(retrieved[0], equalTo(expected[0]));
        assertThat(retrieved[1], equalTo(expected[1]));
        assertThat(retrieved[2], equalTo(expected[2]));
    }

    @Test
    public void testSetCell()
    {
        aBoard.setCell(1, 'X');
        aBoard.setCell(9, 'O');

        assertThat(aBoard.getCell(1), is('X'));
        assertThat(aBoard.getCell(9), is('O'));

        Board.Pair[] retrieved = aBoard.get3Cells(1, 5, 9);
        Board.Pair[] expected = {new Board.Pair(0, 'X'),
                                 new Board.Pair(4, '5'),
                                 new Board.Pair(8, 'O')};

        assertThat(retrieved[0], equalTo(expected[0]));
        assertThat(retrieved[1], equalTo(expected[1]));
        assertThat(retrieved[2], equalTo(expected[2]));

        assertThat(aBoard.toString(), not(equalTo(expectedEmptyStr)));
        assertThat(aBoard.toString(), equalTo("X|2|3\n4|5|6\n7|8|O\n"));

        assertFalse(aBoard.isFull());
    }

}

File: Review-04-Unit-Testing/Example-6/tests/examples/test_board.py
from hamcrest import *
import unittest

from examples.board import Board

import copy


class TestBoard(unittest.TestCase):
    """
    1 - Does this piece of code perform the operations
        it was designed to perform?

    2 - Does this piece of code do something it was not
        designed to perform?

    1 Test per mutator
    """
    expectedEmptyStr = "1|2|3\n4|5|6\n7|8|9"

    def setUp(self):

        self.aBoard = Board()

    def testDefaultConstructor(self):

        for expectedChar in range(1, 10):
            assert_that(self.aBoard.getCell(expectedChar),
                        is_(str(expectedChar)))

        assert_that(str(self.aBoard), equal_to(TestBoard.expectedEmptyStr))
        self.assertFalse(self.aBoard.isFull())

        retrieved = self.aBoard.get3Cells(1, 2, 3)
        expected = ((0, '1'), (1, '2'), (2, '3'))

        assert_that(retrieved[0], equal_to(expected[0]))
        assert_that(retrieved[1], equal_to(expected[1]))
        assert_that(retrieved[2], equal_to(expected[2]))

    def testSetCell(self):

        self.aBoard.setCell(1, 'X')
        self.aBoard.setCell(9, 'O')

        assert_that(self.aBoard.getCell(1), is_('X'))
        assert_that(self.aBoard.getCell(9), is_('O'))

        retrieved = self.aBoard.get3Cells(1, 5, 9)
        expected = ((0, 'X'), (4, '5'), (8, 'O'))

        assert_that(retrieved[0], equal_to(expected[0]))
        assert_that(retrieved[1], equal_to(expected[1]))
        assert_that(retrieved[2], equal_to(expected[2]))

        assert_that(str(self.aBoard),
                    is_not(equal_to(TestBoard.expectedEmptyStr)))
        assert_that(str(self.aBoard), equal_to("X|2|3\n4|5|6\n7|8|O"))

        self.assertFalse(self.aBoard.isFull())
